src/main/java/ch/wisv/areafiftylan/products/service/TicketServiceImpl.java

```diff
@@ -198,6 +202,15 @@ public void cancelTicketTransfer(String token) {
         return ticketRepository.findByPickupService_True();
     }
 
+    @Override
+    public Ticket assignTicketToUser(Long ticketId, String username) {
+        Ticket ticket = getTicketById(ticketId);
+        User user = userService.getUserByUsername(username);
+        ticket.setOwner(user);
+        return ticketRepository.save(ticket);
```

Comment:
This ticket is not immediately flushed to the database, when does this happen? I can't remember having specific flush calls at the end of a method.